cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -209,14 +209,19 @@ public void getPreviewLogsNext(HttpRequest request, HttpResponder responder,
                                  @QueryParam("filter") @DefaultValue("") String filterStr,
                                  @QueryParam("format") @DefaultValue("text") String format,
                                  @QueryParam("suppress") List<String> suppress) throws Exception {
-    sendLogs(namespaceId, previewId,
+    sendLogs(responder, namespaceId, previewId,
              info -> doNext(info.getLogReader(), responder, info.getLoggingContext(), maxEvents,
                             fromOffsetStr, escape, filterStr, info.getRunRecord(), format, suppress));
   }
 
-  private void sendLogs(String namespaceId, String previewId, Consumer<LogReaderInfo> logsResponder) throws Exception {
+  private void sendLogs(HttpResponder responder, String namespaceId, String previewId,
+                        Consumer<LogReaderInfo> logsResponder) throws Exception {
     ApplicationId applicationId = new ApplicationId(namespaceId, previewId);
     ProgramRunId programRunId = previewManager.getRunId(applicationId);
+    if (programRunId == null) {
+      responder.sendStatus(HttpResponseStatus.OK);
```

Comment:
Why it is OK instead of NOT_FOUND? The UI can't handle it correctly?

Reply:
Returning `OK` is similar behavior from 6.1.3 https://github.com/cdapio/cdap/blob/v6.1.3/cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/preview/PreviewHttpHandler.java#L227 as well. I updated to return `NOT_FOUND` in between, but seems like its temporary state till the run id is propagated back from preview runner. UI shows NOT_FOUND during that time. Reverting it to make is same as 6.1.3